cohere/client.py

```diff
@@ -765,10 +765,14 @@ def list_datasets(self, dataset_type: str = None, limit: int = None, offset: int
             "offset": offset,
         }
         response = self._request(f"{cohere.DATASET_URL}", method="GET", params=param_dict)
-        return [
-            Dataset.from_dict({"meta": response.get("meta"), **r}, wait_fn=self.wait_for_dataset)
-            for r in response["datasets"]
-        ]
+        return (
+            [
+                Dataset.from_dict({"meta": response.get("meta"), **r}, wait_fn=self.wait_for_dataset)
+                for r in response["datasets"]
+            ]
+            if "datasets" in response
```

Comment:
I would use `for r in response.get('datasets',[])` or `for r in (response.get('datasets') or [])` but that's just a suggestion